sycl/test-e2e/AddressSanitizer/out-of-bounds/buffer/buffer_copy.cpp

```diff
@@ -0,0 +1,40 @@
+// REQUIRES: linux, cpu
+// RUN: %{build} %device_asan_flags -O0 -g -o %t.out
+// RUN: env SYCL_PREFER_UR=1 %{run} not %t.out 2>&1 | FileCheck %s
+// RUN: %{build} %device_asan_flags -O1 -g -o %t.out
+// RUN: env SYCL_PREFER_UR=1 %{run} not %t.out 2>&1 | FileCheck %s
+// RUN: %{build} %device_asan_flags -O2 -g -o %t.out
+// RUN: env SYCL_PREFER_UR=1 %{run} not %t.out 2>&1 | FileCheck %s
+
+#include <sycl/detail/core.hpp>
+
+static const int N = 16;
+
+int main() {
+  sycl::queue q;
+
+  std::vector<int> v(N);
+  for (int i = 0; i < N; i++)
+    v[i] = i;
+
+  {
+    sycl::buffer<int, 1> buf(v.size());
+
+    q.submit([&](sycl::handler &h) {
+       auto A = buf.get_access<sycl::access::mode::write>(h);
+       h.copy(&v[0], A);
```

Comment:
Do we need the same for `fill`/`memset`?

Reply:
Thanks, updated. BTW, the case just wants to test that buffer copy/fill will not crash since in UR side we intercept all buffer related APIs with USM.